dev-utils/update-tav-versions.js

```diff
@@ -150,7 +137,17 @@ async function main() {
     } else if (tavToUpdate && line.startsWith('  versions:')) {
       console.log(`Updating versions of ${tavToUpdate}`);
       const tavVers = tavVersMap.get(tavToUpdate);
-      tavLines[idx] = `  versions: '${tavVers.join(' || ')}'`;
+      if (tavVers.length > 0) {
+        tavLines[idx] = `  versions: '${tavVers.join(' || ')}'`;
+      }
+    } else if (tavToUpdate && line.startsWith('  - versions:')) {
+      console.log(`Updating versions of ${tavToUpdate}`);
+      const tavMatrix = tavVersMap.get(tavToUpdate);
+      const tavVers = tavMatrix.shift();
+
+      if (tavVers.length > 0) {
+        tavLines[idx] = `  - versions: '${tavVers.join(' || ')}'`;
+      }
     }
```

Comment:
It might be helpful to have a guard that will error out if there are any unused `tavVersMap` entries. Could this happen if there is a block with multiple versions sections, but the first field is not `versions`, e.g. something like:

```yaml
mysql2:
  - foo: bar
    versions: '>=1.0.0 <1.6.0'
    commands: "..."
```